### ISSUE TITLE
Incremental annotation processor support broken with JDK 11+ modules

### ISSUE THREAD

**Issue by graemerocher** [gradle/gradle#11059]
Since adding support for Gradle incremental annotation processors this has now broken Micronaut's support for JDK 11+ when using the module system.

### Expected Behavior

Should compile

### Current Behavior

Fails compilation with:

```
Caused by: java.lang.IllegalArgumentException: Unexpected element micronaut.test.modules.issue
        at org.gradle.api.internal.tasks.compile.processing.ElementUtils.getTopLevelTypeName(ElementUtils.java:71)
        at org.gradle.api.internal.tasks.compile.processing.ElementUtils.getTopLevelTypeNames(ElementUtils.java:47)
        at org.gradle.api.internal.tasks.compile.processing.AggregatingProcessingStrategy.recordAggregatedTypes(AggregatingProcessingStrategy.java:60)
        at org.gradle.api.internal.tasks.compile.processing.AggregatingProcessingStrategy.recordProcessingInputs(AggregatingProcessingStrategy.java:46)
        at org.gradle.api.internal.tasks.compile.processing.DynamicProcessingStrategy.recordProcessingInputs(DynamicProcessingStrategy.java:52)
        at org.gradle.api.internal.tasks.compile.processing.DynamicProcessor.process(DynamicProcessor.java:51)
```

### Context

JDK 11 Module support with Micronaut is broken for Gradle builds and currently only works with Maven.

### Steps to Reproduce 

See example: https://github.com/graemerocher/micronaut-test-modules-issue

Run: gradle cJ --stacktrace

### Your Environment

JDK 11+ and any version of Gradle

**Comment by graemerocher** [comment]
Still present in 5.6.3

**Comment by msgilligan** [comment]
Still present in Gradle 6.0 RC2

**Comment by chrispoulsen** [comment]
And sadly still present in 6.0...

**Comment by big-guy** [comment]
@graemerocher could you give `6.1-20191212033525+0000` a try (this is the same version as I linked in the other issue)?

**Comment by graemerocher** [comment]
Fixed in 6.1
